mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/quota/CassandraQuotaCurrentValueDao.java

```diff
@@ -0,0 +1,88 @@
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.quota;
+
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.bindMarker;
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.insertInto;
+import static com.datastax.oss.driver.api.querybuilder.QueryBuilder.selectFrom;
+import static com.datastax.oss.driver.api.querybuilder.relation.Relation.column;
+
+import javax.inject.Inject;
+
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor;
+import org.apache.james.core.quota.QuotaComponent;
+import org.apache.james.core.quota.QuotaComponentFactory;
+import org.apache.james.core.quota.QuotaCurrentValue;
+import org.apache.james.core.quota.QuotaType;
+import org.apache.james.core.quota.QuotaTypeFactory;
+import org.apache.james.mailbox.cassandra.table.CassandraQuotaCurrentValue;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.cql.PreparedStatement;
+import com.datastax.oss.driver.api.querybuilder.insert.Insert;
+import com.datastax.oss.driver.api.querybuilder.select.Select;
+
+import reactor.core.publisher.Mono;
+
+public class CassandraQuotaCurrentValueDao {
+
+    private final CassandraAsyncExecutor queryExecutor;
+    private final PreparedStatement getQuotaCurrentValueStatement;
+    private final PreparedStatement setQuotaCurrentValueStatement;
+    private final QuotaComponentFactory quotaComponentFactory;
+    private final QuotaTypeFactory quotaTypeFactory;
+
+    @Inject
+    public CassandraQuotaCurrentValueDao(CqlSession session, QuotaComponentFactory quotaComponentFactory, QuotaTypeFactory quotaTypeFactory) {
+        this.queryExecutor = new CassandraAsyncExecutor(session);
+        this.getQuotaCurrentValueStatement = session.prepare(getQuotaCurrentValueStatement().build());
+        this.setQuotaCurrentValueStatement = session.prepare(setQuotaCurrentValueStatement().build());
+        this.quotaComponentFactory = quotaComponentFactory;
+        this.quotaTypeFactory = quotaTypeFactory;
+    }
+
+    Mono<QuotaCurrentValue> getQuotaCurrentValue(String identifier, QuotaComponent quotaComponent, QuotaType quotaType) {
+        return queryExecutor.executeSingleRow(getQuotaCurrentValueStatement.bind(identifier, quotaComponent.asString(), quotaType.asString()))
+            .map(row -> QuotaCurrentValue.of(row.get(CassandraQuotaCurrentValue.IDENTIFIER, String.class), quotaComponentFactory.parse(row.get(CassandraQuotaCurrentValue.QUOTA_COMPONENT, String.class)),
+                quotaTypeFactory.parse(row.get(CassandraQuotaCurrentValue.QUOTA_TYPE, String.class)), row.get(CassandraQuotaCurrentValue.CURRENT_VALUE, Long.class)));
+    }
```

Comment:
Not readable a first site (6am here)

Please try extractoring IE row -> POJO convertion